google/cloud/storage/emulator/database.py

```diff
@@ -269,13 +279,38 @@ def __to_serializeable_retry_test(cls, retry_test):
             "completed": retry_test["completed"],
         }
 
+    def insert_supported_methods(self, methods):
+        self.supported_methods.extend(methods)
+
     def get_retry_test(self, retry_test_id):
         retry_test = self.retry_tests.get(retry_test_id)
         if retry_test is None:
             utils.error.notfound("Retry Test %s" % retry_test_id, context=None)
         return self.__to_serializeable_retry_test(retry_test)
 
+    def __validate_instructions(self, instructions):
+        expected_instructions_format = [
+            utils.common.retry_return_error_code.pattern,
+            utils.common.retry_return_error_connection.pattern,
+            utils.common.retry_return_error_after_bytes.pattern,
+        ]
+        search_expression = re.compile(
+            "(%s)" % ")|(".join(expected_instructions_format)
+        )
+        expected_schema = Schema(
```

Comment:
Have not seen this technique using a schema in python before; it feels more reminiscent of unmarshaling in Go. Pretty cool.